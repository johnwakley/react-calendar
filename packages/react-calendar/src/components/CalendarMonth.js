/**
 * MIT License
 *
 * Copyright (c) 2017 johnwakley
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 *
 */

// @flow

import React from 'react'
import styled, { ThemeProvider } from 'styled-components'
import { ical as theme } from '../themes'
import type { CalendarMonthEnum } from 'react-calendar-core'
import {
  calendarMonth,
  DatePresenter,
  dateStyles,
  media,
  DEFAULT_LOCALE
} from 'react-calendar-core'
import CalendarDay from './CalendarDay'

const StyledMonth = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: 1em;
  /* stylelint-disable */
  ${media.tablet`margin: 0.5em;`}
  ${media.mobile`margin: 0.25em;`}
  /* stylelint-enable */
  font-family: ${props => props.theme.typography.familySansSerif};
  font-size: ${props => props.theme.typography.size6};
  /* stylelint-disable */
  ${media.tablet`font-size: ${props => props.theme.typography.size7};`}
  ${media.mobile`font-size: ${props => props.theme.typography.size8};`}
  /* stylelint-enable */
  background-color: white;
`

const MonthHeader = styled.div`
  flex: 1;
  padding: 0.5em;
  /* stylelint-disable */
  ${media.desktop`padding-left: 5%;`}
  ${media.tablet`padding-left: 4%;`}
  ${media.mobile`padding-left: 3%;`}
  /* stylelint-enable */
  text-align: left;
  font-size: ${props => props.theme.typography.size4};
  /* stylelint-disable */
  ${media.tablet`font-size: ${props => props.theme.typography.size6};`}
  ${media.mobile`font-size: ${props => props.theme.typography.size7};`}
  /* stylelint-enable */
  color: ${props => props.theme.colors.orange};
`

const StyledWeek = styled.div`
  display: flex;
  flex: 1;
`

const Week = (props: any) => {
  return (
    <StyledWeek>
      {props.children}
    </StyledWeek>
  )
}

const WeekDays = Week

export const CalendarMonth = ({
  year,
  month,
  locale = DEFAULT_LOCALE
}: {
  year: number,
  month: CalendarMonthEnum,
  locale: string
}) => {
  const cal = calendarMonth(year, month)

  return (
    <ThemeProvider theme={theme}>
      <StyledMonth>
        <MonthHeader>{cal.monthName}</MonthHeader>
        <WeekDays>
          {cal.weekdaysShort.map(day =>
            <CalendarDay key={day} dateStyle={dateStyles.DAY_NAME}>
              {day}
            </CalendarDay>
          )}
        </WeekDays>
        {cal.weeks.map(week =>
          <Week key={cal.weeks.indexOf(week)}>
            {week.map((date: Date) => {
              const datePresenter = new DatePresenter(date, month)
              const day = date.getDate()
              return (
                <CalendarDay key={day} dateStyle={datePresenter.displayStyle()}>
                  {day}
                </CalendarDay>
              )
            })}
          </Week>
        )}
      </StyledMonth>
    </ThemeProvider>
  )
}
