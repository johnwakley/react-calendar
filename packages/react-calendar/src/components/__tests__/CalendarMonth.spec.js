import React from 'react'
import renderer from 'react-test-renderer'
import { CalendarMonth } from '../CalendarMonth'

describe('CalendarMonth', () => {
  it('should show month', () => {
    const tree = renderer
      .create(<CalendarMonth year={2017} month={5} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
