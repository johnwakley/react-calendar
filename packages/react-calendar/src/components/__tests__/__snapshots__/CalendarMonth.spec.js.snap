// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarMonth should show month 1`] = `
<div
  className="sc-bdVaJa ffSamm"
>
  <div
    className="sc-bwzfXH kosksu"
  >
    June
  </div>
  <div
    className="sc-htpNat dPjGwt"
  >
    <div
      className="sc-bxivhb iayOwP"
    >
      <div
        className="sc-ifAKCX iQLrSg"
      >
        Sun
      </div>
    </div>
    <div
      className="sc-EHOje jytZOR"
    >
      <div
        className="sc-bZQynM uZQAu"
      >
        Mon
      </div>
    </div>
    <div
      className="sc-gzVnrw kCYczj"
    >
      <div
        className="sc-htoDjs gdNlEO"
      >
        Tue
      </div>
    </div>
    <div
      className="sc-dnqmqq mWOgS"
    >
      <div
        className="sc-iwsKbI bqTilp"
      >
        Wed
      </div>
    </div>
    <div
      className="sc-gZMcBi jcdOCC"
    >
      <div
        className="sc-gqjmRU ghUhZO"
      >
        Thu
      </div>
    </div>
    <div
      className="sc-VigVT bQdnss"
    >
      <div
        className="sc-jTzLTM gHEqvc"
      >
        Fri
      </div>
    </div>
    <div
      className="sc-fjdhpX hELONV"
    >
      <div
        className="sc-jzJRlG dQOLlQ"
      >
        Sat
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat dPjGwt"
  >
    <div
      className="sc-cSHVUG yggOO"
    >
      <div
        className="sc-kAzzGY fZEISB"
      >
        28
      </div>
    </div>
    <div
      className="sc-chPdSV cwnwCX"
    >
      <div
        className="sc-kgoBCf kYDghQ"
      >
        29
      </div>
    </div>
    <div
      className="sc-kGXeez dRohDh"
    >
      <div
        className="sc-kpOJdX IhOEz"
      >
        30
      </div>
    </div>
    <div
      className="sc-dxgOiQ ijcqxc"
    >
      <div
        className="sc-ckVGcZ dzTxss"
      >
        31
      </div>
    </div>
    <div
      className="sc-jKJlTe htjdIU"
    >
      <div
        className="sc-eNQAEJ kmQFZs"
      >
        1
      </div>
    </div>
    <div
      className="sc-hMqMXs bZdHKA"
    >
      <div
        className="sc-kEYyzF dBTyik"
      >
        2
      </div>
    </div>
    <div
      className="sc-kkGfuU cYltK"
    >
      <div
        className="sc-iAyFgw hxtSmo"
      >
        3
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat dPjGwt"
  >
    <div
      className="sc-hSdWYo fcJpRz"
    >
      <div
        className="sc-eHgmQL lfQTT"
      >
        4
      </div>
    </div>
    <div
      className="sc-cvbbAY fXxuJk"
    >
      <div
        className="sc-jWBwVP bRuuUL"
      >
        5
      </div>
    </div>
    <div
      className="sc-brqgnP lghFPn"
    >
      <div
        className="sc-cMljjf hNKuJe"
      >
        6
      </div>
    </div>
    <div
      className="sc-jAaTju iejBVB"
    >
      <div
        className="sc-jDwBTQ fBCTDr"
      >
        7
      </div>
    </div>
    <div
      className="sc-gPEVay SzdHS"
    >
      <div
        className="sc-iRbamj EkUDy"
      >
        8
      </div>
    </div>
    <div
      className="sc-jlyJG cNedej"
    >
      <div
        className="sc-gipzik ioylVB"
      >
        9
      </div>
    </div>
    <div
      className="sc-csuQGl fNcaxg"
    >
      <div
        className="sc-Rmtcm dPOtbv"
      >
        10
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat dPjGwt"
  >
    <div
      className="sc-bRBYWo bKWnOi"
    >
      <div
        className="sc-hzDkRC KFkdX"
      >
        11
      </div>
    </div>
    <div
      className="sc-jhAzac gTqVHN"
    >
      <div
        className="sc-fBuWsC dOzGyN"
      >
        12
      </div>
    </div>
    <div
      className="sc-fMiknA cBHnrJ"
    >
      <div
        className="sc-dVhcbM iJSmpZ"
      >
        13
      </div>
    </div>
    <div
      className="sc-eqIVtm bzvgUb"
    >
      <div
        className="sc-fAjcbJ btczTp"
      >
        14
      </div>
    </div>
    <div
      className="sc-caSCKo knfCeb"
    >
      <div
        className="sc-gisBJw dBltYR"
      >
        15
      </div>
    </div>
    <div
      className="sc-kjoXOD kKPDNR"
    >
      <div
        className="sc-cHGsZl dnLELM"
      >
        16
      </div>
    </div>
    <div
      className="sc-TOsTZ dJzjlS"
    >
      <div
        className="sc-kgAjT kAvHLT"
      >
        17
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat dPjGwt"
  >
    <div
      className="sc-cJSrbW jGUHhF"
    >
      <div
        className="sc-ksYbfQ sgNiU"
      >
        18
      </div>
    </div>
    <div
      className="sc-hmzhuo cZbfLL"
    >
      <div
        className="sc-frDJqD ersTXg"
      >
        19
      </div>
    </div>
    <div
      className="sc-kvZOFW huEEBe"
    >
      <div
        className="sc-hqyNC iRcNuU"
      >
        20
      </div>
    </div>
    <div
      className="sc-jbKcbu hnNsle"
    >
      <div
        className="sc-dNLxif irEBvS"
      >
        21
      </div>
    </div>
    <div
      className="sc-jqCOkK ikBCdt"
    >
      <div
        className="sc-uJMKN cvXCy"
      >
        22
      </div>
    </div>
    <div
      className="sc-bbmXgH bURDwc"
    >
      <div
        className="sc-gGBfsJ bymMbm"
      >
        23
      </div>
    </div>
    <div
      className="sc-jnlKLf dBvFru"
    >
      <div
        className="sc-fYxtnH bJcwtB"
      >
        24
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat dPjGwt"
  >
    <div
      className="sc-tilXH kYnhbq"
    >
      <div
        className="sc-hEsumM fhsndH"
      >
        25
      </div>
    </div>
    <div
      className="sc-ktHwxA bktDiu"
    >
      <div
        className="sc-cIShpX kHmBFZ"
      >
        26
      </div>
    </div>
    <div
      className="sc-kafWEX efmRLN"
    >
      <div
        className="sc-feJyhm jzzLLc"
      >
        27
      </div>
    </div>
    <div
      className="sc-iELTvK kaAfsl"
    >
      <div
        className="sc-cmTdod eWHawO"
      >
        28
      </div>
    </div>
    <div
      className="sc-jwKygS dijJLx"
    >
      <div
        className="sc-btzYZH gkcPKt"
      >
        29
      </div>
    </div>
    <div
      className="sc-lhVmIH fFQCAB"
    >
      <div
        className="sc-bYSBpT brvqYH"
      >
        30
      </div>
    </div>
    <div
      className="sc-elJkPf fwCNpN"
    >
      <div
        className="sc-jtRfpW dMPdas"
      >
        1
      </div>
    </div>
  </div>
</div>
`;
